### DISCUSSION TITLE
Can't figure out what's the difference of three service modes

### DISCUSSION THREAD

**Discussion by doublesixwings** [shadowsocks/shadowsocks-android#2718] [Q&A]
Please read FAQ then answer these questions before submitting your issue. Thanks!

### Environment

* Android version: 7.1.1
* Device: Sony Xperia Z5 Premium Dual (E6883),  Hongkong version, with Google services itself
* Shadowsocks version: v4.5.6
* Last version that did not exhibit the issue (if applicable):

### Configuration

_Put an `x` inside the [ ] that applies._

* [x] IPv4 server address
* [ ] IPv6 server address
* [x] Client IPv4 availability
* [ ] Client IPv6 availability
* Encrypt method:
* Route
  * [ ] All
  * [ ] Bypass LAN
  * [ ] Bypass China
  * [x] Bypass LAN & China
  * [ ] GFW List
  * [ ] China List
  * [ ] Custom rules
* [ ] IPv6 route
* [ ] Apps VPN mode
  * [ ] Bypass mode
* Remote DNS: 8.8.8.8
* [ ] DNS Forwarding
* Plugin configuration (if applicable): kcptun
* [ ] Auto Connect
* [ ] TCP Fast Open
* If you're not using VPN mode, please supply more details here:

### What did you do?
I tried three service modes in wifi and 4G(China Mobile) respectively. There are two apps I tested with, Youtube and Google Now. I found in wifi environment, ‘only proxy’ and ‘transparent proxy’ worked perfectly in both of the two apps, 'vpn' went well in Youtube but failed in Google Now. In 4G environment, 'vpn' went well in both of the two apps, ‘only proxy’ and ‘transparent proxy’ failed in both of the two apps. Is this normal? I must use different service mode in wifi or 4G environment to make everything goes well?

### What did you expect to see?
I hope one service mode can go well both in wifi and 4G  environment.

### What did you see instead?
As mentioned above.

PS. I have shadowsocks installed on my router, so I can fuck GFW without using shadowsocks-android in my phone, I use it because of the kcptun client plugin, since my router isn't capable of running this cpu-consuming software.

**Comment by Mygod** [comment]
Does it work without kcptun?

**Comment by doublesixwings** [comment]
Thanks for the reply. I just tested working without kcptun. Here's the situation: In wifi environment, all of these three modes worked perfectly in both youtube and google now apps, in 4G environment, 'vpn' went well in both of the two apps, ‘only proxy’ and ‘transparent proxy’ failed in both of the two apps. The strange thing is, when I turned back to work with kcptun again (I havn't use ss-android since I posted this issue) , I got exactly the same results as working without kcptun, which means I could choose the ‘vpn’ mode to make everyting goes well both in wifi and 4G environment, so my problem is solved,  although I don't know how this happened. By the way,  I do make some config changes to my vps and router to make them work more fluently after I posted this issue, maybe the result could be attributed to those actions. Sincerely thanks for your contribution.

**Comment by Mygod** [comment]
Proxy and transproxy mode is only for *advanced* user. If you don't know what those are, you probably don't need them.

**Comment by FossPrime** [comment]
Transparent Proxy Mode:
Allows you to give other apps access to a Shadowsocks server's services. The example in the FAQ is useful for people with rooted Android phones who want more performance or wish to use the proxy without appearing to have a VPN turned on.
